lookup/dotenv.py

```diff
@@ -45,6 +45,9 @@ def run(self, terms, variables=None, **kwargs):
             if params['key'] is None:
                 ret.append(data)
             else:
-                ret.append(data[params['key']])
+                try:
+                    ret.append(data[params['key']])
+                except (KeyError) as e:
```

Comment:
```suggestion
                except Exception as e:
```

why not wrap all exception that happen?

Reply:
cause that's not what i see other plugins do :)
you should handle exceptions that can occur from erroneous plugin usage, AFAIK.

Reply:
changed

Reply:
Ok, in the example they also wrap Exception: https://docs.ansible.com/ansible/latest/dev_guide/developing_plugins.html#raising-errors

Btw there are also more specific Exceptions available: https://github.com/ansible/ansible/blob/devel/lib/ansible/errors/__init__.py#L202